rpc/server.go

```diff
@@ -141,20 +139,22 @@ func (s *Server) ServeCodec(codec ServerCodec) {
 	s.codecsMu.Lock()
 	if atomic.LoadInt32(&s.run) != 1 { // server stopped
 		s.codecsMu.Unlock()
-		return
+		return &shutdownError{}
 	}
 	s.codecs.Add(codec)
```

Comment:
If we're doing a single shot request, shouldn't the codec rather be defer closed instead of saving it into the set?